go/vt/vtgate/evalengine/arithmetic.go

```diff
@@ -212,6 +223,22 @@ func NullsafeCompare(v1, v2 sqltypes.Value) (int, error) {
 	if isByteComparable(v1) && isByteComparable(v2) {
 		return bytes.Compare(v1.ToBytes(), v2.ToBytes()), nil
 	}
+	if v1.IsText() && v2.IsText() && collationID != collations.Unknown {
+		collation := collations.FromID(collationID)
+		if collation == nil {
+			return 0, UnsupportedCollationError{
+				ID: collationID,
+			}
+		}
+		switch result := collation.Collate(v1.ToBytes(), v2.ToBytes(), false); {
```

Comment:
we don't really need to switch on this, right? we should be able to just return the value from `collation.Collate` here, I think

Reply:
`Collate` returns a positive or negative or zero value and `NullSafeCompare` needs to return -1,0 or 1.